From 27eceb2e4ecbf2e31940b67cee682a6935f61e6c Mon Sep 17 00:00:00 2001
From: Tom Gundersen <teg@jklm.no>
Date: Mon, 27 Oct 2014 17:55:03 +0100
Subject: [PATCH] static-nodes: indicate that creation of static nodes should
 only happen at boot

udev will only manage static nodes that exist at the time udev is started, so
creating static nodes later on will likely not behave as expected. In
particular, recreating the static nodes at run-time will reset any permissions
udev may have applied to the nodes at boot.

See <https://bugzilla.redhat.com/show_bug.cgi?id=1147248> and the discussion
following <http://permalink.gmane.org/gmane.comp.sysutils.systemd.devel/23795>.

Note that this requires (the yet to be released) systemd v217 or a backport
of systemd patch 8c94052ee543c3598a3c7b0c46688150aa2c6168.
---
 tools/static-nodes.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/static-nodes.c b/tools/static-nodes.c
index d1e3b6e..8d2356d 100644
--- a/tools/static-nodes.c
+++ b/tools/static-nodes.c
@@ -96,7 +96,7 @@ static int write_tmpfiles(FILE *out, char modname[], char devname[], char type,
 			return EXIT_FAILURE;
 	}
 
-	ret = fprintf(out, "%c /dev/%s 0600 - - - %u:%u\n",
+	ret = fprintf(out, "%c! /dev/%s 0600 - - - %u:%u\n",
 		      type, devname, maj, min);
 	if (ret < 0)
 		return EXIT_FAILURE;
-- 
2.1.2

